core/primitives/src/runtime/parameter_table.rs

```diff
@@ -60,10 +60,54 @@ impl ParameterValue {
     }
 }
 
+fn format_number(mut n: u64) -> String {
+    let mut parts = Vec::new();
+    while n >= 1000 {
+        parts.push(format!("{:03?}", n % 1000));
+        n /= 1000;
+    }
+    parts.push(n.to_string());
+    parts.reverse();
+    parts.join("_")
+}
+
+impl core::fmt::Display for ParameterValue {
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+        match self {
+            ParameterValue::U64(v) => write!(f, "{}", format_number(*v)),
+            ParameterValue::Rational { numerator, denominator } => {
+                write!(f, "{numerator} / {denominator}")
+            }
+            ParameterValue::Fee { send_sir, send_not_sir, execution } => {
+                write!(
+                    f,
+                    r#"
+- send_sir:     {}
+- send_not_sir: {}
+- execution:    {}"#,
+                    format_number(*send_sir),
+                    format_number(*send_not_sir),
+                    format_number(*execution)
+                )
+            }
+            ParameterValue::String(v) => write!(f, "{v}"),
+        }
+    }
+}
+
 pub(crate) struct ParameterTable {
     parameters: BTreeMap<Parameter, ParameterValue>,
 }
 
+impl core::fmt::Display for ParameterTable {
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+        for (key, value) in &self.parameters {
+            write!(f, "{key:40}{value}\n")?
```

Comment:
Or maybe just align everything to the right? :thinking: 
```suggestion
            write!(f, "{key:40}{value:>20}\n")?
```

Reply:
Not sure why, but this doesn't seem to work - this alignment is just ignored. I've added the alignment for the individual types for now.

Reply:
Ah, hm I think it's because this is calling the custom `Display` of `ParameterValue` which does not respect alignment hints. For this to work, you would have to check the formatter flag [f.align()](https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.align), I think. (I have never tried this myself.)

But the current solution is simpler and works just as well.